engine-precompiles/src/random.rs

```diff
@@ -39,9 +39,10 @@ impl Precompile for RandomSeed {
         &self,
         input: &[u8],
         target_gas: Option<EthGas>,
-        _context: &Context,
+        context: &Context,
         _is_static: bool,
     ) -> EvmPrecompileResult {
+        utils::validate_no_value_attached_to_precompile(context.apparent_value.as_u128())?;
```

Comment:
Why not to change the `validate_no_value_attached_to_precompile` function like this:
```rust
pub fn validate_no_value_attached_to_precompile(value: U256) -> Result<(), ExitError> {
     if value > U256::zero() {
         // don't attach native token value to that precompile
         return Err(ExitError::Other(Borrowed("ATTACHED_VALUE_ERROR")));
     }
     Ok(())
 }
```
In such way you don't need to call `.as_u128()`.

Reply:
Yes, I was going to change it. But I am still waiting a confirmation from @joshuajbouw about whether it is the expected fix or not.